src/pubsub/ua_pubsub_networkmessage.h

```diff
@@ -239,9 +179,87 @@ typedef struct {
     UA_ByteString signature;
 } UA_NetworkMessage;
 
+/**********************************************/
+/*          Network Message Offsets           */
+/**********************************************/
+
+/* Offsets for buffered messages in the PubSub fast path. */
+typedef enum {
+    UA_PUBSUB_OFFSETTYPE_DATASETMESSAGE_SEQUENCENUMBER,
+    UA_PUBSUB_OFFSETTYPE_NETWORKMESSAGE_SEQUENCENUMBER,
+    UA_PUBSUB_OFFSETTYPE_TIMESTAMP_PICOSECONDS,
+    UA_PUBSUB_OFFSETTYPE_TIMESTAMP,     /* source pointer */
+    UA_PUBSUB_OFFSETTYPE_TIMESTAMP_NOW, /* no source */
+    UA_PUBSUB_OFFSETTYPE_PAYLOAD_DATAVALUE,
+    UA_PUBSUB_OFFSETTYPE_PAYLOAD_VARIANT,
+    UA_PUBSUB_OFFSETTYPE_PAYLOAD_RAW,
+	/* For subscriber RT */
+	UA_PUBSUB_OFFSETTYPE_PUBLISHERID,
+	UA_PUBSUB_OFFSETTYPE_WRITERGROUPID,
+    UA_PUBSUB_OFFSETTYPE_DATASETWRITERID
+    /* Add more offset types as needed */
+} UA_NetworkMessageOffsetType;
+
+typedef struct {
+    UA_NetworkMessageOffsetType contentType;
+    union {
+        union {
```

Comment:
Is the union right or should it be a struct?

Reply:
I did not modify this structure. I just moved it according to the usage.

Seeing this, I think struct would be the correct one and I did not modify this here as I am not so sure.

Reply:
Please change it to struct in this PR, i think the inside union was an mistake and the change to struct shouldn’t break any code.